### PR TITLE
Usbsubsystem

### PR DESCRIPTION
**Pull request by madhurimaparuchuri**

### CODE REVIEW

**Comment by carlescufi** [comment]
@madhurimaparuchuri was this PR a mistake? I don't see any commits here that are not already in other PRs?

**Comment by carlescufi** [comment]
> @madhurimaparuchuri was this PR a mistake? I don't see any commits here that are not already in other PRs?

Oh right I see the last commit now. Why don't you add it to the ongoing PR:
https://github.com/zephyrproject-rtos/zephyr/pull/45601

**Comment by madhurimaparuchuri** [comment]
> > @madhurimaparuchuri was this PR a mistake? I don't see any commits here that are not already in other PRs?
> 
> Oh right I see the last commit now. Why don't you add it to the ongoing PR: #45601

I see the PR: #45601 is big enough already and in the review process from 2+ weeks, if I add my commit on top of it, it may extend the time for it to be merged in mainline, hence raised separate draft pull request. I am checking with owner of PR anyways if I can merge it. Thank you.

File: cmake/modules/vif.cmake [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Executes generate_vif.py to create vif.xml.
+
+set(CMD_GEN_VIF ${PYTHON_EXECUTABLE} ${GEN_VIF_SCRIPT}
+        --dts ${ZEPHYR_DTS_PRE}
+        --vif-out ${VIF_XML}.new
+        --bindings-dirs ${DTS_BINDINGS}
+        --compatible ${DTS_COMPATIBLE}
+        )
+
+execute_process(
```
Comment (yperess) [inline cmake/modules/vif.cmake:23]: Does this work? I thought that `execute_process` will happen at config time for CMake. I think you need to create a target:

```
add_custom_target(
  vif_target
  COMMAND ${CMD_GEN_VIF}
  DEPENDS ${ZEPHYR_DTS_PRE}
  WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
  RESULT_VARIABLE reg
)
```

You might need to tweak the `DEPENDS` line there.

**Review by yperess — CHANGES_REQUESTED** [review]

File: cmake/modules/vif.cmake (hunk 1 above)
Comment (madhurimaparuchuri) [reply cmake/modules/vif.cmake:23]: I actually followed the way gen_defines.py generates zephyr.dts from zephyr.dts.pre at the same folder vif.xml is being generated : https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/modules/dts.cmake#L210

**Pull request closed without merging**